internal/imex/imex.go

```diff
@@ -46,8 +46,10 @@ func GetChannels(config *spec.Config, devRoot string) (Channels, error) {
 		channelName := "channel" + id
 		path := filepath.Join("/dev/nvidia-caps-imex-channels", channelName)
 		channel := Channel{
-			ID:       id,
-			Path:     path,
+			ID:   id,
+			Path: path,
+			// TODO: HostPath is not the correct name for this property.
+			// We should deal with this consistently.
 			HostPath: filepath.Join(devRoot, path),
```

Comment:
Is this a different `devRoot` or is this a different `path`? At a superficial glance I don't see how the value for the change in [internal/plugin/server.go](https://github.com/NVIDIA/k8s-device-plugin/pull/996/files#diff-1855ca77ecb9dcdd1c59676ab0ad69676b1c219d76484174bbec4187252de5f2) is different than this...

Reply:
The only difference here is the comment. I can update the comment to be clearer.

Reply:
expanded the comment and moved it to where we made the change.

Reply:
I understand that the comment is the only difference here -- what I'm trying to understand is how this (already existing assignment):
```
HostPath: filepath.Join(devRoot, path)
```
is differnt from the updated assignement we are doing in [internal/plugin/server.go](https://github.com/NVIDIA/k8s-device-plugin/pull/996/files#diff-1855ca77ecb9dcdd1c59676ab0ad69676b1c219d76484174bbec4187252de5f2):
```
HostPath:    filepath.Join(devRoot, channel.Path),
```
Is it a different `devRoot` is the `path` in the first case somehow different than `channel.Path` in the second (even though `channel.Path` seems to just be assigned to the same `path` that is passed in.

Reply:
Got it -- the new comment clarifies...

Reply:
As I called out offline and in the updated comment. The `devRoot` here and in the device node injection code in server.go are different paths. The `devRoot` in our updated assignment is set from the `NVIDIA_DRIVER_ROOT` (`/`) wherease the value here is `/host` which is the driver root in the CONTAINER.